sdk/core/src/services/github/service.rs

```diff
@@ -0,0 +1,141 @@
+
+use crate::services::github::error::Error;
+use crate::services::github::client::{
+    Client as GitHubClient,
+    Repo,
+};
+
+/// Definition of the GitHubProvider
+#[derive(Debug)]
+pub struct GitHubService {
+    org: String,
+    client: GitHubClient,
+}
+
+/// Impl for GitHubSbomProvider
+///
+impl GitHubService {
+    /// new method sets the organization for the struct
+    ///
+    pub fn new(org: String, pat: String) -> Self {
+        let client = GitHubClient::new(pat);
+        GitHubService {
+            org,
+            client,
+        }
+    }
+
+    /// Use the GitHub Client to get all of the repositories
+    /// from the GitHub Organization
+    pub(crate) async fn get_repos(&self) -> Result<Vec<Repo>, Error> {
```

Comment:
Once we add the Harbor `Repository` entity, we want to make sure to refactor this to return harbor types.

Reply:
We'll do this in a subsequent PR per our conversation.